### PR TITLE
Fix #2143  Adding link for leaderboard

### LINKED ISSUE #2143: Create a unique link for each of the entry on the leaderboard
## Current Scenario

If a participant wants to share their rank in a challenge on twitter/facebook then they have to share the entire leaderboard link.

## Expected scenario
There should be a unique link for each entry and upon opening, it should scroll to that particular entry and highlight it.

### PR DESCRIPTION
**Pull request by sarthak212**
Fix #2143

@RishabhJain2018 can you please review this?

### CODE REVIEW

File: frontend/src/js/controllers/challengeCtrl.js [hunk 1]
```diff
@@ -6,9 +6,9 @@
         .module('evalai')
         .controller('ChallengeCtrl', ChallengeCtrl);
 
-    ChallengeCtrl.$inject = ['utilities', 'loaderService', '$scope', '$state', '$http', '$stateParams', '$rootScope', 'Upload', '$interval', '$mdDialog', 'moment'];
+    ChallengeCtrl.$inject = ['utilities', 'loaderService', '$scope', '$state', '$http', '$stateParams', '$rootScope', 'Upload', '$interval', '$mdDialog', 'moment','$location','$anchorScroll'];
 
-    function ChallengeCtrl(utilities, loaderService, $scope, $state, $http, $stateParams, $rootScope, Upload, $interval, $mdDialog, moment) {
+    function ChallengeCtrl(utilities, loaderService, $scope, $state, $http, $stateParams, $rootScope, Upload, $interval, $mdDialog, moment,$location,$anchorScroll) {
```
Comment (Ram81) [inline frontend/src/js/controllers/challengeCtrl.js]: add space after `,`

File: frontend/src/css/modules/leaderboard.scss [hunk 2]
```diff
@@ -3,3 +3,10 @@
 .horizontal-scroll {
 	overflow-x: auto;
 }
+.challengeleaderboardlink{
+	color:black;
+	text-decoration: none;
+}
+.lightgrey{
```
Comment (Ram81) [inline frontend/src/css/modules/leaderboard.scss]: Add newline before every new class

File: frontend/src/js/controllers/challengeCtrl.js [hunk 3]
```diff
@@ -55,7 +55,29 @@
...
+        $scope.beforeLink=$location.absUrl();
+        var reversestring=$scope.beforeLink.split("").reverse();
+        var indexOfString=reversestring.indexOf('/');
+        $scope.beforeLink=reversestring.slice(indexOfString-1,).reverse().join("");
+
+        // watch when there is change in url
+        $scope.$watch(function(){
+          return $location.absUrl();
+        },
+      function(value){
```
Comment (Ram81) [inline frontend/src/js/controllers/challengeCtrl.js]: Indent properly

**Comment by sarthak212** [comment]
@Ram81 you can see the feature in gif.

 leaderboard

**Comment by sarthak212** [comment]
@RishabhJain2018 I have made changes according to Bug. Please review

**Pull request closed without merging**